internal/providers/node_runtime_deps_info.bzl

```diff
@@ -114,9 +114,17 @@ def run_node(ctx, inputs, arguments, executable, **kwargs):
                 env[var] = ctx.configuration.default_shell_env[var]
     env["BAZEL_NODE_MODULES_ROOT"] = _compute_node_modules_root(ctx)
 
+    # ctx.actions.run accepts both lists and a depset for inputs. Coerce the original inputs to a
+    # depset if they're a list, so that extra inputs can be combined in a performant manner.
+    inputs_depset = depset(transitive = [
+        depset(direct = inputs) if type(inputs) == "list" else inputs,
+        extra_inputs,
+        depset(direct = [modules_manifest]),
```

Comment:
nit: `depset(transitive=[depset(direct=[one_thing])])` can always be simplified to `depset([one_thing])`